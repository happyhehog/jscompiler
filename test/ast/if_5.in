if (a) { thenFunc(); } else elseFunc();