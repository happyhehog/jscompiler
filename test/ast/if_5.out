ProgramNode (1:0, 1:38)
    IfStatementNode (1:0, 1:37)
    --> test expression:
        IdentifierNode (1:4, 1:4) : { name: a }
    --> consequent:
        BlockStatementNode (1:7, 1:21)
            CallExpressionNode (1:9, 1:18)
            --> callable:
                IdentifierNode (1:9, 1:9) : { name: thenFunc }
    --> alternate:
        CallExpressionNode (1:28, 1:37)
        --> callable:
            IdentifierNode (1:28, 1:28) : { name: elseFunc }
